rosidl_generator_cpp/rosidl_generator_cpp/__init__.py

```diff
@@ -176,7 +176,106 @@ def primitive_value_to_cpp(type_, value):
     assert False, "unknown primitive type '%s'" % type_.type
 
 
+def default_cpp_value_from_type(type_):
+    if type_ == 'string':
+        return ''
```

Comment:
Should this be `return '""'`?

Reply:
As it stands, no.  The reason is that these values will all be passed through `primitive_value_to_cpp` (either directly or indirectly through `value_to_cpp`) below, which will make these "real" C++ code, including the quotes and any trailing modifiers.